### PR TITLE
fix(backstage-plugin-buildkite): wait step support

### LINKED ISSUE #536: Buildkite Plugin: 404 on "wait: ~" steps in Pipeline
When using the `@roadiehq/backstage-plugin-buildkite`, navigating to a specific build path (enabled in https://github.com/RoadieHQ/roadie-backstage-plugins/commit/1a2208929e2d05006a0f2da9f7f6bb4691272f1f) causes 404 error notifications when the Buildkite Pipeline being loaded includes `wait: ~` steps.

Example Buildkite Pipeline File
```yaml
steps:
  - label: Build
    commands:
      - make build

  - wait: ~

  - label: After Wait To Show 404 Issue
    agents:
        queue: v1
    commands:
      - echo "demo"
```

## Expected Behavior
The Buildkite Plugin Specific Build page for an Entity should properly display a build step for `wait: ~` steps

## Current Behavior
Buildkite Pipelines that include "wait" steps cause 404 errors

image

## Steps to Reproduce

These steps assume a configured environment and are run locally

1. Create Buildkite pipeline that includes a wait step (see example above)
2. Import Entity with proper Buildkite annotation for that pipeline
3. Navigate to `CI/CD` tab on Entity Page
4. Click into a specific build
5. See 404 where `wait` step should be

## Possible Solution

Handle Pipeline steps that do not have log output / labels (potential fix for the 404 errors)

## Context

* There are no server logs or console logs that show anything but a 404 error

## Your Environment

- Backstage version: `1.1.1`
- `@backstage/plugin-catalog-react` version: `1.0.1`
- `@backstage/integration` version: `1.1.0`
- `@roadiehq/backstage-plugin-buildkite` version: `2.0.3`

### PR DESCRIPTION
**Pull request by alecjacobs5401**
Resolves #536

* adds `BuildkiteBuildStep` component to handle rendering different types of job types
* fixes a typo in a type check
* adds testing of the `BuildKiteBuildView` page

Forgive the slew of formatting changes, my `prettier` auto corrected a few files

### Before
before-image

### After
after-image

#### :heavy_check_mark: Checklist

- [x] Added tests for new functionality and regression tests for bug fixes
- [x] Added changeset (run `yarn changeset` in the root)
- [x] Screenshots of before and after attached (for UI changes)
- [x] Added or updated documentation (if applicable) **N/A**

### CODE REVIEW

File: plugins/frontend/backstage-plugin-buildkite/src/components/BuildKiteBuildView/BuildKiteBuildView.tsx [hunk 1]
```diff
@@ -124,30 +126,34 @@ const ActionsList: FC<{ jobs: BuildkiteJob[]}> = ({
...
+  });
 
   useEffect(() => {
     fetchBuildData();
   }, [fetchBuildData]);
 
   useEffect(() => {
     let timer: ReturnType<typeof setTimeout>;
-    if(value?.state === 'running' || value?.state === 'schedudled') {
+    if (value?.state === 'running' || value?.state === 'scheduled') {
```
Comment (alecjacobs5401) [inline plugins/frontend/backstage-plugin-buildkite/src/components/BuildKiteBuildView/BuildKiteBuildView.tsx:153]: typo I spotted while in here

File: plugins/frontend/backstage-plugin-buildkite/src/components/types.ts [hunk 2]
```diff
@@ -35,6 +35,6 @@ export type TableProps = {
 
 export type BuildkiteJob = {
   state: string;
-  log_url: string;
+  log_url?: string;
```
Comment (alecjacobs5401) [inline plugins/frontend/backstage-plugin-buildkite/src/components/types.ts:38]: made this optional given that `waiter` types **don't** have a `log_url`

File: plugins/frontend/backstage-plugin-buildkite/src/components/BuildKiteBuildView/BuildKiteBuildView.test.tsx [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+  [buildKiteApiRef, new BuildkiteApi({ discoveryApi, fetchApi })],
+];
+
+describe('BuildkiteBuildView', () => {
+  const worker = setupServer();
+  setupRequestMockHandlers(worker);
+
+  beforeEach(() => jest.resetAllMocks());
+
+  it('should display a page build and job information', async () => {
```
Comment (rikez) [inline plugins/frontend/backstage-plugin-buildkite/src/components/BuildKiteBuildView/BuildKiteBuildView.test.tsx:63]: nit: found a bit difficult to understand this test. Is it possible to assign the steps to variables, idk. just an idea to improve readability.

**Review by rikez — APPROVED** [review]
lgtm, left 1 comment but not a blocker.

File: plugins/frontend/backstage-plugin-buildkite/src/components/BuildKiteBuildView/BuildKiteBuildView.test.tsx (hunk 3 above)
Comment (asaf-erlich) [reply plugins/frontend/backstage-plugin-buildkite/src/components/BuildKiteBuildView/BuildKiteBuildView.test.tsx:63]: Sometimes with these you can break it up to separate `it`s. But easier for you to know if that's very hard to do.

**Review by asaf-erlich — COMMENTED** [review]
LGTM overall

**Review by kissmikijr — APPROVED** [review]
nice work, thank you!

**Pull request merged**